### PR TITLE
OSOE-751: Upgrade to Orchard Core 1.8

### PR DESCRIPTION
**Pull request by Psichorex**
[OSOE-751](https://lombiq.atlassian.net/browse/OSOE-751)

[OSOE-751]: https://lombiq.atlassian.net/browse/OSOE-751?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Piedone — APPROVED** [review]

File: Lombiq.Hosting.Azure.ApplicationInsights/Services/TelemetryFilter.cs [hunk 1]
```diff
@@ -10,19 +10,16 @@
 
 namespace Lombiq.Hosting.Azure.ApplicationInsights.Services;
 
-public class TelemetryFilter : ITelemetryProcessor
+public partial class TelemetryFilter : ITelemetryProcessor
 {
     private static readonly List<Regex> _expectedErrors =
     [
         // Using this generic error message to filter because it is not possible to filter only those cases where it is
```
Comment (Piedone) [inline Lombiq.Hosting.Azure.ApplicationInsights/Services/TelemetryFilter.cs]: Move this comment below to the method.

File: Lombiq.Hosting.Azure.ApplicationInsights/Services/TelemetryFilter.cs [hunk 2]
```diff
@@ -99,4 +96,14 @@ private DependencyTelemetry GetDependencyTelemetryFailureToIgnore(ITelemetry ite
         // Name property value could be different depending on the environment, so using the Data property instead.
         return dependency.Data.Contains(mediaBlobStorageContainerName) ? dependency : null;
     }
+
+    [GeneratedRegex(@"Azure\.RequestFailedException: The specified container already exists\.",
+        RegexOptions.Compiled,
+        matchTimeoutMilliseconds: 1000)]
+    private static partial Regex RequestFailedRegex();
```
Comment (Piedone) [inline Lombiq.Hosting.Azure.ApplicationInsights/Services/TelemetryFilter.cs]:
```suggestion
    private static partial Regex AzureRequestFailedRegex();
```

File: Lombiq.Hosting.Azure.ApplicationInsights/Services/TelemetryFilter.cs [hunk 3]
```diff
@@ -99,4 +96,14 @@ private DependencyTelemetry GetDependencyTelemetryFailureToIgnore(ITelemetry ite
         // Name property value could be different depending on the environment, so using the Data property instead.
         return dependency.Data.Contains(mediaBlobStorageContainerName) ? dependency : null;
     }
+
+    [GeneratedRegex(@"Azure\.RequestFailedException: The specified container already exists\.",
+        RegexOptions.Compiled,
```
Comment (Piedone) [inline Lombiq.Hosting.Azure.ApplicationInsights/Services/TelemetryFilter.cs]: `Compiled` is [not used in these cases](https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-source-generators#:~:text=The%20RegexOptions.Compiled%20flag%20is%20ignored%20by%20the%20source%20generator%2C%20thus%20making%20it%20no%20longer%20needed%20in%20the%20source%20generated%20version.).

**Review by Piedone — CHANGES_REQUESTED** [review]

File: Lombiq.Hosting.Azure.ApplicationInsights/Services/TelemetryFilter.cs (hunk 1 above)
Comment (Psichorex) [reply Lombiq.Hosting.Azure.ApplicationInsights/Services/TelemetryFilter.cs]: Do you mean the comment at line 17 to be moved where the Regex is being declared?

Comment (Piedone) [reply Lombiq.Hosting.Azure.ApplicationInsights/Services/TelemetryFilter.cs]: Yes, since that's where it makes understanding easier.

**Review by Piedone — APPROVED** [review]

**Review by Piedone — APPROVED** [review]

**Review by Piedone — APPROVED** [review]

**Pull request merged**